docs/README.md

```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
+    kubectl create namespace istio-system
+    kubectl create namespace istio-cni
+    ```
+
+2. Create the `Istio-cni` resource
+  
+      ```bash
+      cat <<EOF | kubectl apply -f -
+      apiVersion: operator.istio.io/v1alpha1
+      kind: IstioCNI
+      metadata:
+        name: default
+      spec:
+        version: v1.21.0
+        namespace: istio-cni
+      EOF
+      ```
+
+3. Create the `Istio` resource
+
+    ```bash
+    cat <<EOF | kubectl apply -f-
+    apiVersion: operator.istio.io/v1alpha1
+    kind: Istio
+    metadata:
+      name: default
+    spec:
+      namespace: istio-system
+      updateStrategy:
+        type: RevisionBased
+        inactiveRevisionDeletionGracePeriodSeconds: 30
+      version: v1.21.0
+    EOF
+    ```
+
+4. Confirm the control plane is installed and is using the desired version.
+
+    ```bash
+    $ kubectl get istio -n istio-system
+    NAME      READY   STATUS    IN USE   VERSION   AGE
+    default   True    Healthy   True     v1.21.0   2m
+    ```
+
+5. Get the `istiorevision` name
+
+    ```bash
+    $ kubectl get istiorevision -n istio-system
+    NAME              READY   STATUS    IN USE   VERSION   AGE
+    default-v1-21-0   True    Healthy   False    v1.21.0   114s
+    ```
+
+6. Create `bookinfo` namespace and label it with the revision name
+
+    ```bash
+    kubectl create namespace bookinfo
+    kubectl label namespace bookinfo istio.io/rev=default-v1-21-0
+    ```
+
+7. Deploy bookinfo application
+
+    ```bash
+    kubectl apply -n bookinfo -f https://raw.githubusercontent.com/istio/istio/release-1.22/samples/bookinfo/platform/kube/bookinfo.yaml
+    ```
+
+8. Confirm proxy sidecar injection version match the control plane version
+
+    ```bash
+    kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
+    docker.io/istio/examples-bookinfo-details-v1:1.19.1
+    docker.io/istio/proxyv2:1.21.0
+    ```
+
+9. Update the control plane to a new version
+
+    ```bash
+    cat <<EOF | kubectl apply -f-
+    apiVersion: operator.istio.io/v1alpha1
+    kind: Istio
+    metadata:
+      name: default
+    spec:
+      namespace: istio-system
+      updateStrategy:
+        type: RevisionBased
+        inactiveRevisionDeletionGracePeriodSeconds: 30
+      version: v1.21.2
+    EOF
+    ```
+
+10. Verify the `Istio` and `IstioRevision` resources. There will be a new revision created with the new version.
+
+    ```bash
+    $ kubectl get istio -n istio-system
+      NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   VERSION   AGE
+      default   2           2       1        Healthy           v1.21.2   23m
+    $ kubectl get istiorevision -n istio-system
+      NAME              READY   STATUS    IN USE   VERSION   AGE
+      default-v1-21-0   True    Healthy   True     v1.21.0   27m
+      default-v1-21-2   True    Healthy   False    v1.21.2   4m45s
+    ```
+
+11. Confirm there is two control plane pods running, one for each revision
```

Comment:
```suggestion
11. Confirm there are two control plane pods running, one for each revision
```

Reply:
@sridhargaddam We do have tests that check the grace period. Are you sure they're getting deleted immediately and not after 30s (the default grace period)?

Reply:
@luksa I was following the steps mentioned in this PR and I'm fairly sure of seeing this behavior. 

However, I believe the reason is that when the initial Istio object was created at Line 248, it was given the name default. When it is updated at Line 302, we are merely updating the same Istio object. 

Shouldn't we use a different name for `Istio` objects when using revisions?

Reply:
No, using the same object is correct. The operator takes care of creating two `IstioRevision` objects underneath.

Reply:
There was some confusion on my end. I misread it as 30 mins and not seconds. We are good. Thanks guys!